### PR TITLE
Fixes #123 `Feature/ShareButton`

### LINKED ISSUE #123: [FEATURE] Adding a functional share button
### Description

I would like to contribute by making a functional share button, clicking on which, we will have options to share the post via Whatsapp, Linkedin, Telegram, etc. having the link of the post and some information about the post.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sarkartanmay393**
### Screenvideo

https://user-images.githubusercontent.com/84321236/224238085-221c729c-b552-4c75-98d6-4e85cd72b8e2.mp4

### CODE REVIEW

**Comment by sarkartanmay393** [comment]
@narayan954 It's done ✅

File: src/App.jsx [hunk 1]
```diff
@@ -228,6 +239,100 @@ function App() {
...
+          style={{
+            backgroundColor: "var(--bg-color)",
+            padding: "20px",
+            textAlign: "center",
+            color: "var(--color)",
+            border: "2px solid var(--color)",
+          }}
+        >
+          <img
+            src="https://user-images.githubusercontent.com/27727921/185767526-a002a17d-c12e-4a6a-82a4-dd1a13a5ecda.png"
```
Comment (narayan954) [inline src/App.jsx:257]: try the logo from assets/logo.png , fix alt to be dummygram, also the classname should modal signup img

File: src/App.jsx [hunk 2]
```diff
@@ -228,6 +239,100 @@ function App() {
...
+          <p
+            style={{
+              fontSize: "1rem",
+              color: "var(--color)",
+            }}
+          >
+            Share Post
+          </p>
+
+          <div className="mx-2">
```
Comment (narayan954) [inline src/App.jsx:271]: icons are not dark mode proof. Please fix them

**Review by narayan954 — CHANGES_REQUESTED** [review]
Looks great,but need to fix those minor stuffs. Also, I'm thinking of converting these chunks of code into separate components. So could you please create a component for this share modal and I'll handle the rest chunk later on. App.js is getting unnecessarily complex for newcomers and separating this logic could help.

File: src/App.jsx (hunk 2 above)
Comment (sarkartanmay393) [reply src/App.jsx:271]: Sure, and also I would like to create separate components. Let me create the issue after it and will ask for an assignment.

Comment (narayan954) [reply src/App.jsx:271]: right, it'll be of great help!

**Comment by sarkartanmay393** [comment]
@narayan954 Check once more

**Comment by narayan954** [comment]
> @narayan954 Check once more

are you sure, you have pushed any changes?

**Comment by narayan954** [comment]
> > @narayan954 Check once more
> 
> are you sure, you have pushed any changes?

I had rebased this branch, maybe that'd the reason

**Comment by sarkartanmay393** [comment]
@narayan954 Can you see the changes now ?

**Comment by sarkartanmay393** [comment]
@narayan954  I can't the vercel deployments.

**Comment by narayan954** [comment]
> @narayan954 Can you see the changes now ?

Yes, I can, reviewing it.

**Review by narayan954 — APPROVED** [review]
looks good to me! :)

**Review by narayan954 — APPROVED** [review]
lgtm

**Pull request merged**